tensorflow/compiler/xla/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -251,6 +251,13 @@ namespace wrap {
 
 #endif
 
+#define HIP_CHECK(status)                                                                          \
+    if (status != hipSuccess) {                                                                    \
+        std::cout << "Got Status: " << status << " at Line: " << __LINE__ << std::endl;            \
+        exit(0);                                                                                   \
```

Comment:
exit(0) is definitely great, can we do better? Propagate the status?

Reply:
Thanks! 

"Propagate the status", do you mean by using **CHECK_EQ** like this?

```
CHECK_EQ(hipStreamSynchronize(hip_stream), hipSuccess) << "Failed to set hipStream.";
```

Reply:
Well ideally no: ideally it should propagate all the way to the caller, with macros like TF_RETURN_IF_ERROR

Reply:
What's done in the CUDA backend for those cases?

Reply:
It used a similar macros like TF_RETURN_IF_ERROR as the following

```
#define RETURN_IF_CUDNN_ERROR(expr)                                     \
  do {                                                                  \
    cudnnStatus_t _status = (expr);                                     \
    if (!ABSL_PREDICT_TRUE(_status == CUDNN_STATUS_SUCCESS)) {          \
      std::ostringstream oss;                                           \
      oss << CudnnStatusToString(_status) << "\nin " << __FILE__ << "(" \
          << __LINE__ << "): '" << #expr << "'";                        \
      return port::Status(port::error::UNKNOWN, oss.str());             \
    }                                                                   \
  } while (false)
```

Do you expect a similar macro for ROCm like this?

```
#define RETURN_IF_ROCMDNN_ERROR(expr)                                     \
  do {                                                                    \
    hipdnnStatus_t _status = (expr);                                          \
    if (!ABSL_PREDICT_TRUE(_status == miopenStatusSuccess)) {             \
      std::ostringstream oss;                                             \
      oss << ToString(_status) << "\nin " << __FILE__ << "("              \
          << __LINE__ << "): '" << #expr << "'";                        \
      return port::Status(port::error::UNKNOWN, oss.str());             \
    }                                                                   \
  } while (false)```

Reply:
Yes, that would make sense.

Reply:
Apologies for my last comment, actually in cuda side (cu_dnn.cc), they used CHECK_EQ to handle the cuda stream part.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc#L220-L221

And this HIP_CHECK is for handling ignored warning (hipSuccess) from hipStreamSynchronize(), which is used in two void functions (find and trim) to work with PoolingWorkspaceCache. 

- find() https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/stream_executor/rocm/rocm_dnn.cc#L4111
- trim() https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/stream_executor/rocm/rocm_dnn.cc#L4142

This is because MIOpen cached kernels to speed up whilst cudnn probably implemented it inside.

hipStreamSynchronize() also doesn't have DNN status here. All DNN related status in either rocm_dnn.cc or cuda_dnn.cc are already propagated by miOpenStatus_t and cudnnStatus_t, respectively. They are different to hipSuccess.

So I think use the existing CHECK_EQ with a related logging message is a more reasonable way to do, rather than creating a new HIP_CHECK.
```
CHECK_EQ(hipStreamSynchronize(hip_stream), hipSuccess) << "Failed to sync hipStream.";
```

Hope it make sense.